docs/docs/create-transformer-plugin.md

```diff
@@ -2,7 +2,54 @@
 title: Create a Transformer Plugin
 ---
 
-This is a stub. Help our community expand it.
+Transformer plugins are responsible for "transforming" data provided by
+[source plugins](/docs/create-source-plugin) into new nodes and/or node fields.
 
-Please use the [Gatsby Style Guide](/docs/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+## Transformer plugin example
+
+The [`gatsby-transform-remark`](/packages/gatsby-transform-remark/) plugin
+"transforms" markdown nodes (fetched for example from the filesystem source plugin) into html and provides additional node details such as for example the `timeToRead` the `tableOfContents` and the `excerpt`.
+
+## What do transformer plugins do?
+
+Each transformer plugin is responsible for "enhancing" the existing source nodes they
+are provided to provide additional information. Therefore, you'll often use both source plugins and transformer plugins in your Gatsby sites. Transformer plugins can be heavily opinionated since they choose they way they transform their source.
+
+This loose coupling between the data source and transformer plugins allow Gatsby
+site builders to quickly assemble complex data transformation pipelines with
+little work.
+
+## What does the code look like?
+
+Just like a source plugin, a transformer plugin is a normal NPM package. It has a `package.json` file with optional
+dependencies as well as a `gatsby-node.js` file where you implement Gatsby's Node.js
+APIs. Transformer plugins make use of the `setFieldsOnGraphQLNodeType` function and they commonly provide this functionality in the `extend-node-type` file.
+
+Your `gatsby-node.js` should look something like:
+
+```javascript
```

Comment:
We're using a new way of indicated code snippet files:

```
```javascript:title=gatsby-node.js
```

Would you mind updated your Markdown document to include something similar to this ^ for each code snippet?

Reply:
Yeah no problem. just done this.